packages/tabs/src/tab-nav.vue

```diff
@@ -0,0 +1,376 @@
+<script lang='ts'>
+import { h, defineComponent, ref, inject, computed, onUpdated, onMounted, onBeforeUnmount } from 'vue'
+import { addResizeListener, removeResizeListener } from '@element-plus/utils/resize-event'
+import { firstUpperCase } from '@element-plus/utils/util'
+import TabBar from './tab-bar.vue'
+import { NOOP } from '@vue/shared'
+
+export default defineComponent({
+  name: 'ElTabNav',
+  components: {
+    TabBar,
+  },
+  props: {
+    panes: {
+      type: Array,
+      default: () => ([]),
+    },
+    currentName: {
+      type: String,
+      default: '',
+    },
+    editable: Boolean,
+    onTabClick: {
+      type: Function,
+      default: NOOP,
+    },
+    onTabRemove: {
+      type: Function,
+      default: NOOP,
+    },
+    type: {
+      type: String,
+      default: '',
+    },
+    stretch: Boolean,
+  },
+  setup(props) {
+    const rootTabs = inject('rootTabs', { props })
+
+    const scrollable = ref(false)
+    const navOffset = ref(0)
+    const isFocus = ref(false)
+    const focusable = ref(true)
+
+    const navScroll$ = ref(null)
+    const nav$ = ref(null)
+    const el$ = ref(null)
```

Comment:
These should have a expected interface, `ref<null | HTMLElement>`

Reply:
```ts
type RefElement = Nullable<HTMLElement>
const el$ = ref<RefElement>(null)
```